rohmu/object_storage/sftp.py

```diff
@@ -192,15 +192,17 @@ def store_file_from_memory(self, key, memstring, metadata=None, cache_control=No
     def store_file_from_disk(self, key, filepath, metadata=None, multipart=None, cache_control=None, mimetype=None):
         with open(filepath, "rb") as fh:
             self.store_file_object(key=key, fd=fh, cache_control=cache_control, metadata=metadata, mimetype=mimetype)
-            self.notifier.object_created(key=key, size=os.path.getsize(fh))
+            self.notifier.object_created(key=key, size=os.fstat(fh.fileno()).st_size)
 
     # pylint: disable=unused-argument
     def store_file_object(self, key, fd, *, cache_control=None, metadata=None, mimetype=None, upload_progress_fn=None):
         target_path = self.format_key_for_backend(key.strip("/"))
+        notify_size = [0]
 
         self.log.debug("Store path: %r", target_path)
 
         def wrapper_upload_progress_fn(bytes_written, total_bytes):  # pylint: disable=unused-argument
+            notify_size[0] = bytes_written
```

Comment:
This should accumulate the written bytes, not overwrite the previous value.

Reply:
Really? Paramiko docs say that the first arg is the number of bytes sent so far, and callback implementation here https://github.com/paramiko/paramiko/blob/main/paramiko/sftp_client.py#L675 seems to agree with that, already doing the accumulation of individual writes.